sky/provision/azure/config.py

```diff
@@ -92,10 +94,19 @@ def bootstrap_instances(
                 retry += 1
                 continue
             raise
+        except azure.exceptions().ClientAuthenticationError as e:
+            message = (
+                'Failed to authenticate with Azure. Please check your Azure '
+                f'credentials. Error: {common_utils.format_exception(e)}'
+            ).replace('\n', ' ')
+            logger.error(message)
+            raise exceptions.NoClusterLaunchedError(message) from e
```

Comment:
Q: why do we need a new exception?

Reply:
See https://github.com/skypilot-org/skypilot/pull/4023#discussion_r1797411690